packages/package_info_plus/package_info_plus/lib/package_info_plus.dart

```diff
@@ -119,4 +119,26 @@ class PackageInfo {
   String toString() {
     return 'PackageInfo(appName: $appName, buildNumber: $buildNumber, packageName: $packageName, version: $version, buildSignature: $buildSignature, installerStore: $installerStore)';
   }
+
+  Map<String, dynamic> _toMap() {
+    final values = <String, dynamic>{
+      'appName': appName,
+      'buildNumber': buildNumber,
+      'packageName': packageName,
+      'version': version,
+    };
+
+    if (buildSignature.isNotEmpty) {
+      values['buildSignature'] = buildSignature;
+    }
+
+    if (installerStore != null && installerStore!.isNotEmpty) {
+      values['installerStore'] = installerStore;
+    }
+
+    return values;
+  }
```

Comment:
How about this?

```suggestion
  Map<String, dynamic> _toMap() {
    return {
      'appName': appName,
      'buildNumber': buildNumber,
      'packageName': packageName,
      'version': 'version',
      if (buildSignature.isNotEmpty) 'buildSignature': buildSignature,
      if (installerStore?.isNotEmpty ?? false) 'installerStore': installerStore,
    };
  }
```

Reply:
The way it works is the same. Personally I tend to avoid conditions along the same lines in favor of more visible ones. But no problem.

Reply:
In my opinion, the more concise version appears to be more commonly used these days.
People familiar with Dart will be able to read it quickly either way.